daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SpeedyTest.scala

```diff
@@ -222,6 +222,14 @@ class SpeedyTest extends WordSpec with Matchers {
     }
   }
 
+  "type abstraction" should {
+    "evaluate when using (un uninhabited) type of kind '* -> nat'" in {
+      val e = e"""/\(f : * -> nat). ADD_NUMERIC @(f Unit)"""
+      val x = eval(e, anyPkgs)
+      x shouldBe 'right
```

Comment:
Maybe it would be good to be more specific on how `x` looks like. Right now, I have no idea what it looks like.

Reply:
What is `x` here? Does the type lambda get stripped out & its body is evaluated?

What is the semantics of the program below?

```
let g = /\(n: nat). error "here"
in /\(f: * -> nat). g (f Unit)
```

(I'm concerned this will blow up the interpreter!)

Reply:
Or something like this,

```
let g = /\(n: nat). \(x:Int64). \(y:Int64). x
in /\(f: * -> nat). g (f Unit) 10 20
```

Is this the same as `/\(f: * -> nat). 10` or `/\(f: * -> nat). \(y: Int64). 20`, or something else?

I don't think it's a good idea to allow these `* -> nat` type variables.

Reply:
ftr we came up with the term 

```
let g: forall(n:nat). Int64 = /\(n: nat). 1
in /\(f: * -> nat). ADD_INT64 10 (g @(f Unit))
```

which crashes the interpreter with a panic, because it violates type preservation.

This puts into question the value of "saving" `* -> nat` altogether.

Reply:
I suggest you report this counter example in #6598 and we close this PR.